axelor-production/src/main/resources/views/PurchaseOrder.xml

```diff
@@ -43,4 +49,75 @@
     <domain>:id member of self.purchaseOrderSet</domain>
   </action-view>
 
+  <action-group name="action-group-purchase-order-partner-onchange"
+    id="action-group-production-purchase-order-partner-onchange">
+    <action name="action-purchase-order-record-partner"/>
+    <action name="action-purchase-order-method-get-stock-location"
+      if="__config__.app.isApp('supplychain') &amp;&amp; (tradingName?.receiptDefaultStockLocation == null || stockLocation == null)"/>
+    <action name="action-purchase-order-method-get-from-stock-location"
+      if="__config__.app.isApp('supplychain')"/>
+    <action name="action-purchase-order-attrs-hide-contact-partner"/>
+    <action name="action-purchase-order-method-fill-company-bank-details"/>
+    <action name="action-purchase-order-method-fill-price-list"/>
+    <action name="action-purchase-order-attrs-interco"/>
+    <action name="action-purchase-order-record-interco"/>
+    <action name="action-purchase-order-record-set-group-product-on-printing"/>
+    <action name="action-purchase-order-method-empty-fiscal-position-if-not-compatible"/>
+    <action name="action-purchase-order-method-fiscal-position-on-change"/>
+    <action name="action-purchase-order-method-compute"/>
+    <action name="action-purchase-order-record-set-type-select"/>
+  </action-group>
+
+
+  <action-record name="action-purchase-order-record-set-type-select"
+    model="com.axelor.apps.purchase.db.PurchaseOrder">
+    <field name="typeSelect" expr="eval: 1"
+      if="!supplierPartner || !supplierPartner.isSubcontractor"/>
+    <field name="typeSelect" expr="eval: 2"
+      if="supplierPartner &amp;&amp; supplierPartner.isSubcontractor"/>
+  </action-record>
+
+  <search-filters name="purchase-order-filters"
```

Comment:
is there a way to avoid code duplication with search filters ? It would be nice because it is duplicating a lot...

Reply:
I don't know if there is a way, I tried to only add what I needed instead of duplicating but it does override everything so I can't do that.. Search-filter property on action view looks like it can only take 1 value also